cmake/modules/FindARIA.cmake

```diff
@@ -0,0 +1,70 @@
+# Filename: FindARIA.cmake
+#
+# Usage:
+#   find_package(ARIA [REQUIRED] [QUIET])
+#
+# Once done this will define:
+#   ARIA_FOUND       - system has ARIA MAgicCrypt0
+#   ARIA_INCLUDE_DIR - the include directory containing ARIA
+#   ARIA_LIBRARY     - the path to the libARIA library
+#
+
+set(ARIA_INCLUDE_DIR)
+set(ARIA_LIB_FILE)
+
+# when debugging cmake, ARIA_DIR environment variable can be manually set here:
+# set(ENV{ARIA_DIR} "/mnt/c/workspace/MagicCrypto")
+# set(ENV{ARIA_DIR} "c:\\workspace\\MagicCrypto")
+
+# Make sure we have a ARIA_DIR encironment variable with the path to MagicCrypto
+if ("$ENV{ARIA_DIR}" STREQUAL "")
+    message(ERROR "ERROR: FindARIA.cmake missing ARIA_DIR value")
```

Comment:
I think this should default to ARIA_DIR=WOLFSSL_ROOT/MagicCrypto instead of giving an error. With autotools, MagicCrypto is explicitly required to be in the root of the wolfSSL directory. We should keep the two build systems in sync. Either have CMake require it in the same spot, or update configure.ac to support the ARIA_DIR. My vote is for the former.

Reply:
I've added additional logic here for _either_ the environment variable _or_ a local copy of MagicCrypto in the wolfSSL repo directory structure.  I've added a new `ARIA_IS_LOCAL` variable and will update this PR description, above. 

Easier said than done, as `target_include_directories()` complains unintuitively if the respective directory is in the current cmake directory structure. See [stackoverflow](https://stackoverflow.com/questions/25676277/cmake-target-include-directories-prints-an-error-when-i-try-to-add-the-source) and the [docs](https://cmake.org/cmake/help/latest/command/target_include_directories.html).

PR update soon.

Reply:
See the [scripts/aria_cmake_build_test.sh](https://github.com/gojimmypi/wolfssl/blob/aria/scripts/aria_cmake_build_test.sh)

Reply:
I like the idea of having the option to set the ARIA directory. Agreed though that it should default to checking the wolfSSL root directory to keep the behavior the same if ARIA_DIR is unset.